schema.zmodel

```diff
@@ -0,0 +1,116 @@
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+// Here's the simplified model
+//   - Permission is changed to enum so that using it doesn't require join models
+
+// Let's use this project to collaborate
+
+enum Permission {
+  ALL
+  CREATE
+  READ
+  UPDATE
+  DELETE
+}
+
+/*
+ * User can be member of many groups, and can operate many posts.
+ */
+model User {
+  id String @id @default(cuid())
+  groups GroupUser[]
+  posts PostUser[]
+  postPermission Permission?
+
+  // a synthesized field to facilitate access policies
+  groupPostPermission Permission? @ignore
```

Comment:
This field is not mapped to the database. It's added to add extra fields to the return value of `auth()` function.

Reply:
The `@ignore` it's really nice to have

Reply:
Ya, it's actually Prisma's built-in feature. It just nicely fit to our needs 😄.

Reply:
> Ya, it's actually Prisma's built-in feature. It just nicely fit to our needs 😄.

Didn't know that 😅